### DISCUSSION TITLE
max_execution_time not working

### DISCUSSION THREAD

**Discussion by olivier-sabban** [brefphp/bref#1811] [Support]
**Description:**

Hi,

I’m working on a project using Bref and need to increase a default execution time of 30 seconds to 60 sec

Here are the steps I’ve taken so far:
	1.	I modified the ﻿serverless.yml file to set the timeout to 30 seconds.
	2.	Despite this, the execution time did not change as expected, so I then modified the ﻿php.ini file.

![image](https://github.com/brefphp/bref/assets/6322328/6d5568d5-c408-4d2a-ad50-a1132001ddaf)
<img width="1400" alt="image" src="https://github.com/brefphp/bref/assets/6322328/f57ebc06-e3e5-45ba-b825-d5b35807392c">
<img width="415" alt="image" src="https://github.com/brefphp/bref/assets/6322328/4925203e-f7e6-48f3-83c1-79d0f44ecc7f">
<img width="975" alt="image" src="https://github.com/brefphp/bref/assets/6322328/df039fec-5cbd-4c1b-9355-99fd4cc0077e">

Currently, the execution can exceed 30 seconds, but I am encountering a CloudFront error if the execution time is between 30 and 60 secondes. Interestingly, the CloudWatch error "The request ***** timed out after 58992 ms. Note: that duration may be lower than the Lambda timeout...."  that I was previously receiving has disappeared when the execution time is between 30 and 60 secondes evenif CloudFront return a error.

<img width="1455" alt="image" src="https://github.com/brefphp/bref/assets/6322328/b41134b5-b85e-403e-99f3-4abcb855eed6">
<img width="979" alt="image" src="https://github.com/brefphp/bref/assets/6322328/c8e4861e-ad46-440c-ae18-22b3a9702e83">

-----

Execution over 60sec. This one is ok :  
<img width="1198" alt="image" src="https://github.com/brefphp/bref/assets/6322328/4674d90c-4f1f-42ed-adf4-ee2bf64549c3">

Could you please advise on what might be causing the CloudFront error and what further steps I might need to take to increase execution time ?

Thank you!

**Comment by mnapoli** [comment]
Just a quick search gave me this

![Screen-001775](https://github.com/brefphp/bref/assets/720328/281b3241-66a6-41cf-8949-66534840d1a0)

have you configured cloudfront at some point?

**Comment by olivier-sabban** [comment]
I changed "Response timeout" in CloudFront > Distributions 

<img width="665" alt="image" src="https://github.com/brefphp/bref/assets/6322328/e71e6c96-40e2-440f-aa8e-1bccfca61345">

I still have error with 503 code :

<img width="295" alt="image" src="https://github.com/brefphp/bref/assets/6322328/a2781910-d501-4ecb-a57e-1579d18e9b26">

No error from lambda in cloudwatch. 

I also tryed to change api gateway timeout. I must not be in the right place; I find it surprising that one cannot have a request for 30 seconds on API GATEWAY.

<img width="706" alt="image" src="https://github.com/brefphp/bref/assets/6322328/44cebd8e-3eea-4a8a-aed3-9019095c5d26">

**Comment by mnapoli** [comment]
Wait are you using API gateway? That cannot work then, API gateway doesn't allow that. Please post all the details of your stack so that we can save some time.

**Comment by olivier-sabban** [comment]
Thank you Matthieu for your help.

It’s a simple serverless Symfony application that you’re brilliantly help us to host on AWS.

Everything works perfectly (assets, php scripts with db, users sessions...) except for scripts that take more than 30 seconds to execute.

composer.json
```
{
    "type": "project",
    "license": "proprietary",
    "minimum-stability": "stable",
    "prefer-stable": true,
    "require": {
        "php": ">=8.2",
        "ext-ctype": "*",
        "ext-iconv": "*",
        "bref/bref": "^2.1",
        "bref/symfony-bridge": "^1.0"
        "doctrine/doctrine-bundle": "^2.11",
        "doctrine/doctrine-migrations-bundle": "^3.3",
        "doctrine/orm": "^2.17",
        "erdum/php-open-ai-assistant-sdk": "^1.0",
        "illuminate/collections": "^10.40",
        "league/commonmark": "^2.4",
        "minwork/array": "^1.13",
        "sentry/sentry-symfony": "^4.13",
        "symfony/console": "6.4.*",
        "symfony/dotenv": "6.4.*",
        "symfony/flex": "^2",
        "symfony/form": "6.4.*",
        "symfony/framework-bundle": "6.4.*",
        "symfony/http-client": "6.4.*",
        "symfony/mailer": "6.4.*",
        "symfony/mailjet-mailer": "6.4.*",
        "symfony/monolog-bundle": "^3.10",
        "symfony/rate-limiter": "6.4.*",
        "symfony/runtime": "6.4.*",
        "symfony/security-bundle": "6.4.*",
        "symfony/serializer": "6.4.*",
        "symfony/twig-bundle": "6.4.*",
        "symfony/validator": "6.4.*",
        "symfony/webpack-encore-bundle": "^2.1",
        "symfony/yaml": "6.4.*",
        "theodo-group/llphant": "^0.3.5",
        "twig/extra-bundle": "^3.8",
        "twig/markdown-extra": "^3.8",
        "twig/twig": "^3.8"
    },
    "require-dev": {
        "doctrine/doctrine-fixtures-bundle": "^3.5"
        "phpunit/phpunit": "^9.5",
        "symfony/browser-kit": "6.4.*",
        "symfony/css-selector": "6.4.*",
        "symfony/debug-bundle": "6.4.*",
        "symfony/maker-bundle": "^1.52",
        "symfony/phpunit-bridge": "^7.0",
        "symfony/stopwatch": "6.4.*",
        "symfony/var-dumper": "6.4.*",
        "symfony/web-profiler-bundle": "6.4.*"
    },
    "config": {
        "allow-plugins": {
            "php-http/discovery": true,
            "symfony/flex": true,
            "symfony/runtime": true
        },
        "sort-packages": true
    },
    "autoload": {
        "psr-4": {
            "App\\": "src/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "App\\Tests\\": "tests/"
        }
    },
    "replace": {
        "symfony/polyfill-ctype": "*",
        "symfony/polyfill-iconv": "*",
        "symfony/polyfill-php72": "*",
        "symfony/polyfill-php73": "*",
        "symfony/polyfill-php74": "*",
        "symfony/polyfill-php80": "*",
        "symfony/polyfill-php81": "*"
    },
    "scripts": {
        "auto-scripts": {
            "cache:clear": "symfony-cmd",
            "assets:install %PUBLIC_DIR%": "symfony-cmd"
        },
        "post-install-cmd": [
            "@auto-scripts"
        ],
        "post-update-cmd": [
            "@auto-scripts"
        ]
    },
    "conflict": {
        "symfony/symfony": "*"
    },
    "extra": {
        "symfony": {
            "allow-contrib": "false",
            "require": "6.4.*"
        }
    }
}
```

serverless.yaml
```
service: test-sleep-60
provider:
  name: aws
  region: eu-west-3
  stage: prod
  runtime: provided.al2
  environment:
    APP_ENV: ${self:provider.stage}
plugins:
  - ./vendor/bref/bref
  - serverless-lift
functions:
  web:
    handler: public/index.php
    timeout: 60
    layers:
      - ${bref:layer.php-82-fpm}
    events:
      - httpApi: '*'
constructs:
  website:
    type: server-side-website
    assets:
      '/build/*': public/build
package:
  patterns:
    - '!node_modules/**'
```
SleepController.php

```
#[Route('/test-timeout', name: 'test_timeout', methods: ['GET'])]
    public function testSleep(Request $request)
    {
        $sleep = $request->query->get('sleep', 20);
        sleep($sleep);
        return $this->json($sleep);
    }
```

Maybe serverless-lift plugin causing the problem ?

**Reply by mnapoli** [reply]
This is API Gateway that prevents any response from taking longer than 30 seconds.

You might want to switch to Lambda Function URL then: https://bref.sh/docs/use-cases/http/advanced-use-cases

Note that this will not be simple, as you will need to create the infrastructure manually and this is not documented (yet… #1625)

**Comment by hschimpf** [comment]
As my knowledge, API Gateway for HTTP API has a hard limit of 30s[^1][^2] that cannot be increased. So if your function runs for more than 30s, API Gateway will respond to CloudFront with a timeout error, even if the function is still running in the background _(idk really if the function is killed by API Gateway, I'll think not)_.

[^1]: https://bref.sh/docs/use-cases/http/advanced-use-cases
[^2]: https://docs.aws.amazon.com/apigateway/latest/developerguide/limits.html#http-api-quotas
